src/CFamily/Rules/Resources.resx

```diff
@@ -117,12 +117,16 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="CFamily_DuplicateLegacyAndNewRuleKey" xml:space="preserve">
+    <value>[CFamily] Rule settings contain entries for both legacy rule key and new rule keys. The legacy value will be ignored. Legacy key: {0}, new key: {1}</value>
```

Comment:
I'm not sure how useful this would be to the end user. If they're using connected mode, then this might confuse them ever further as there is nothing they can do besides upgrading the SQ instance.
I wonder if we should have a wiki page for this or if it's an overkill. Do we have some metrics from cfamily how many users would be impacted by this change?

Reply:
Connected mode: true, it might be confusing if they see the output message. A more noticeable source of confusion is likely to be the Error List, since users connected to "old" SonarQube instances won't see e.g. `cpp:C99CommentUsage` as the rule key, even though that is what they will see in their Quality Profile. We could add a wiki page later if we get community questions about it.

I thought it was better to output something rather than nothing, even though I don't think many users will be affected, and almost none of those who are will notice this in the output window. It was more to help us in answering forum questions if users don't understand what is happening.

Metrics: I haven't checked.